core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -437,7 +437,15 @@ private SearchContext findContext(long id) throws SearchContextMissingException
         if (context == null) {
             throw new SearchContextMissingException(id);
         }
-        return context;
+
+        SearchOperationListener operationListener = context.indexShard().getSearchOperationListener();
+        try {
+            operationListener.validateSearchContext(context);
+            return context;
+        } catch (Exception e) {
```

Comment:
this can only be a runtime exception you can just bubble it up no need to use `convertToRuntime`